gax/src/test/java/com/google/api/gax/rpc/OperationCallableImplTest.java

```diff
@@ -475,9 +475,11 @@ public void testFutureCallPollRPCTimeout() throws Exception {
         OperationTimedPollAlgorithm.create(
             FAST_RECHECKING_SETTINGS
                 .toBuilder()
+                // Update the polling algorithm to set per-RPC timeouts instead of the default zero.
                 .setInitialRpcTimeout(Duration.ofMillis(100))
                 .setMaxRpcTimeout(Duration.ofSeconds(1))
                 .setRpcTimeoutMultiplier(2)
+                .setTotalTimeout(Duration.ofSeconds(5L))
```

Comment:
What is the meaning of totalTimeout in case of LRO? Note that exponential retry algorithm is shared between retries and LRO polling logic. Whatever we do for retries, we must ensure that it does not break LRO. Ideally we want that new logic be completelly disabled fro LRO. I.e. the need to modify LRO tests indicates that there is a breaking behavioral change for LRO, which we should avoid.

Reply:
I had the same thought and I was confused by what this test was doing. 

The [test setup](https://github.com/googleapis/gax-java/blob/master/gax/src/test/java/com/google/api/gax/rpc/OperationCallableImplTest.java#L101-L102) actually sets by default a `totalTimeout` to be `5 ms`, but then in this test increases the `initialRpcTimeout` and `maxRpcTimeout` to values much greater than the existing `totalTimeout` of `5ms`. Per comments in a [generated client](https://github.com/googleapis/java-dataproc/blob/12e77297e659224ded7005f6bfea5df5a35bb6c9/google-cloud-dataproc/src/main/java/com/google/cloud/dataproc/v1/stub/WorkflowTemplateServiceStubSettings.java#L520-L523), the `initialRpcTimeout` and `maxRpcTimeout` should be ignored and set to 0 for LROs? Not really sure what the test is doing here.

Reply:
FWIW we had no GAPIC config for LRO polling `initialRpcTimeout` or `maxRpcTimeout`, so they would never be generated and only ever set by users...and they'd have the same behavior as non-LRO RPCs where a poll could run over the `totalTimeout`

Reply:
I guess this test was testing for the existing incorrect behavior, where the RPC timeout didn't care if it exceeded the `totalTimeout`

Reply:
The `totalTimeout` in the context of LRO is the "total polling timeout" ([gapic config](https://github.com/googleapis/gapic-generator/blob/master/src/main/proto/com/google/api/codegen/v2/config_v2.proto#L575-L576)). So, the duration a "synchronous" LRO call should poll before giving up.

This test was set up weirdly, as I described. PTAL.